autoload/lsp/utils/position.vim

```diff
@@ -89,3 +89,15 @@ function! lsp#utils#position#vim_to_lsp(expr, pos) abort
          \ }
 endfunction
 
+function! lsp#utils#position#is_valid_for_buffer(bufnr, line, col) abort
+    if a:col < 0
+        return 0
+    endif
+
+    let l:line_str = get(getbufline(a:bufnr, a:line, '$'), 0, '')
+    if a:col > strlen(l:line_str)
```

Comment:
better to use strdisplaywidth?

Reply:
@mattn 

`col` parameter of `text_prop()` is counted in bytes. so I used `strlen()`.

> {col} is counted in bytes, use one for the first column.

https://github.com/vim/vim/blob/c0c2c262650103c4a21b64c3246388a350688616/runtime/doc/textprop.txt#L122-L126

Should I rename `lsp#utils#position#is_valid_for_buffer()`?